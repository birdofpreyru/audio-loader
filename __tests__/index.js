/**
 * High-level testing of the library.
 */

import fs from 'fs';
import mockAxios from 'jest-mock-axios';
import audioLoader from '../src';

const MockAudioContext = {
  decodeAudioData: jest.fn((array, onSuccess, onError) => {
    if (array instanceof ArrayBuffer) onSuccess(array.slice(0));
    else onError(Error('Invalid array'));
  }),
};

afterEach(() => {
  mockAxios.reset();
  MockAudioContext.decodeAudioData.mockClear();
});

it('loads MIDI.js soundfont', async () => {
  const URL = '__assets__/piano.js';
  let soundfont = audioLoader(URL, { context: MockAudioContext });
  mockAxios.mockResponseFor({ responseType: 'text', url: URL }, {
    data: fs.readFileSync(`${__dirname}/${URL}`, 'utf8'),
    status: 200,
  });
  soundfont = await soundfont;
  expect(MockAudioContext.decodeAudioData).toHaveBeenCalled();
  expect(soundfont).toMatchSnapshot();
  Object.keys(soundfont).forEach((key) => {
    soundfont[key] = new Uint8Array(soundfont[key]);
  });
  expect(soundfont).toMatchSnapshot();
});

it('refuses to work without AudioContext provided', () => {
  expect(() => audioLoader('__assets__/piano.js'))
    .toThrowErrorMatchingSnapshot();
  expect(() => audioLoader('__assets__/piano.js', {}))
    .toThrowErrorMatchingSnapshot();
});
