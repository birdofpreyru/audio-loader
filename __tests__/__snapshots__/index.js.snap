// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads MIDI.js soundfont 1`] = `
Object {
  "A0": Uint8Array [
    49,
  ],
  "B0": Uint8Array [
    51,
  ],
  "Bb0": Uint8Array [
    50,
  ],
  "C1": Uint8Array [
    67,
    49,
  ],
  "Db1": Uint8Array [
    68,
    98,
    49,
  ],
}
`;

exports[`refuses to work without AudioContext provided 1`] = `"Cannot read property 'context' of undefined"`;

exports[`refuses to work without AudioContext provided 2`] = `"Missing \\"context\\" option"`;
